src/testIntegration/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/prisons/SupportedPrisonsResourceTest.kt

```diff
@@ -1,50 +1,142 @@
+      webTestClient.post()
+        .uri("/prisons/some-prison")
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .exchange()
+        .expectStatus().isUnauthorized
+    }
+
+    @Test
+    fun `forbidden without a valid role`() {
+      webTestClient.post()
+        .uri("/prisons/some-prison")
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .headers(setAuthorisation(user = "AUSER_GEN"))
+        .exchange()
+        .expectStatus().isForbidden
+        .expectBody().jsonPath("$.errorCode.code").isEqualTo(AuthenticationError.code)
+    }
+
+    @Test
+    fun `not found if prison not in prison register`() {
+      PrisonRegisterExtension.prisonRegisterApi.stubGetPrisonNotFound("ABC")
+
+      webTestClient.post()
+        .uri("/prisons/ABC")
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .headers(setAuthorisation(user = "AUSER_GEN", roles = listOf("ROLE_SLM_ADMIN")))
+        .exchange()
+        .expectStatus().isNotFound
+        .expectBody().jsonPath("$.errorCode.code").isEqualTo(NotFound.code)
+    }
+
+    @Test
+    fun `created if the prison has been added`() {
+      PrisonRegisterExtension.prisonRegisterApi.stubGetPrisonOk("ABC")
+
+      webTestClient.post()
+        .uri("/prisons/ABC")
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .headers(setAuthorisation(user = "SOME_USER", roles = listOf("ROLE_SLM_ADMIN")))
+        .exchange()
+        .expectStatus().isCreated
+
+      val supportedPrison = supportedPrisonsRepository.getById("ABC")
+
+      assertThat(supportedPrison)
+        .extracting(SupportedPrison::code, SupportedPrison::active, SupportedPrison::updatedBy)
+        .isEqualTo(listOf("ABC", true, "SOME_USER"))
+      assertThat(supportedPrison.updated).isCloseTo(Instant.now(), within(5, ChronoUnit.SECONDS))
+    }
+
+    @Test
+    fun `created if the prison has been reactivated`() {
+      supportedPrisonsRepository.save(
+        aSupportedPrison(code = "CDE", active = false, updated = Instant.now().minus(1, ChronoUnit.DAYS))
+      )
+      PrisonRegisterExtension.prisonRegisterApi.stubGetPrisonOk("CDE")
+
+      webTestClient.post()
+        .uri("/prisons/CDE")
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .headers(setAuthorisation(user = "SOME_USER", roles = listOf("ROLE_SLM_ADMIN")))
+        .exchange()
+        .expectStatus().isCreated
+
+      val supportedPrison = supportedPrisonsRepository.getById("CDE")
+
+      assertThat(supportedPrison)
+        .extracting(SupportedPrison::code, SupportedPrison::active, SupportedPrison::updatedBy)
+        .isEqualTo(listOf("CDE", true, "SOME_USER"))
+      assertThat(supportedPrison.updated).isCloseTo(Instant.now(), within(5, ChronoUnit.SECONDS))
```

Comment:
Can't use `isEqualTo` because for JPA equality is modelled only on code. The updated date would likely be wrong anyway.